script/release.sh

```diff
@@ -0,0 +1,292 @@
+  echo " tar - you should already have this on your system"
+  echo " hub"
+  echo " github-release"
+  echo " GITHUB_TOKEN in your env variable"
+  echo " "
+  echo "Not only that, but you must have permission for:"
+  echo " Tagging releases within Github"
+  echo ""
+}
+
+requirements() {
+  if [ ! -f /usr/bin/git ] && [ ! -f /usr/local/bin/git ]; then
+    echo "No git. What's wrong with you?"
+    return 1
+  fi
+
+  if [ ! -f /usr/bin/gpg ] && [ ! -f /usr/local/bin/gpg ]; then
+    echo "No gpg. What's wrong with you?"
+    return 1
+  fi
+
+  if [ ! -f $GOPATH/bin/github-release ]; then
+    echo "No $GOPATH/bin/github-release. Please run 'go get -v github.com/aktau/github-release'"
+    return 1
+  fi
+
+  if [ ! -f /usr/bin/hub ]; then
+    echo "Hub needed in order to create the relevant PR's. Please install hub @ https://github.com/github/hub"
+    return 1
+  fi
+
+  if [[ -z "$GITHUB_TOKEN" ]]; then
+    echo "export GITHUB_TOKEN=yourtoken needed for using github-release"
+  fi
+}
+
+# Clone and then change to user's upstream repo for pushing to master / opening PR's :)
+clone() {
+  git clone ssh://git@github.com/$UPSTREAM_REPO/$CLI.git
+  if [ $? -eq 0 ]; then
+        echo OK
+  else
+        echo FAIL
+        exit
+  fi
+  cd $CLI
+  git remote remove origin
+  git remote add origin git@github.com:$ORIGIN_REPO/$CLI.git
+  git checkout -b release-$1
+  cd ..
+}
+
+replaceversion() {
+  cd $CLI
+
+  echo "1. Replaced version in version.go"
+  find . -name 'version.go' -maxdepth 3 -type f -exec sed -i "s/$1/$2/g" {} \;
+
+  echo "2. Replaced README.md versioning"
+  sed -i "s/$1/$2/g" README.md
+  
+  cd ..
+}
+
+changelog() {
+  cd $CLI
+  echo "Getting commit changes. Writing to ../changes.txt"
+  LOG=`git shortlog --email --no-merges --pretty=%s v${1}..`
+  echo -e "\`\`\`\n$LOG\n\`\`\`" > ../changes.txt
+  echo "Changelog has been written to changes.txt"
+  echo "!!PLEASE REVIEW BEFORE CONTINUING!!"
+  echo "Open changes.txt and add the release information"
+  echo "to the beginning of the file before the git shortlog"
+  cd ..
+}
+
+changelog_md() {
+  echo "Generating CHANGELOG.md"
+  CHANGES=$(cat changes.txt)
+  cd $CLI
+  DATE=$(date +"%m-%d-%Y")
+  CHANGELOG=$(cat CHANGELOG.md)
+  HEADER="## $CLI $1 ($DATE)"
+  echo -e "$HEADER\n\n$CHANGES\n\n$CHANGELOG" >CHANGELOG.md
+  echo "Changes have been written to CHANGELOG.md"
+  cd ..
+}
+
+git_commit() {
+  cd $CLI 
+
+  BRANCH=`git symbolic-ref --short HEAD`
+  if [ -z "$BRANCH" ]; then
+    echo "Unable to get branch name, is this even a git repo?"
+    return 1
+  fi
+  echo "Branch: " $BRANCH
+
+  git add .
+  git commit -m "$1 Release"
+  git push origin $BRANCH
+  hub pull-request -b $UPSTREAM_REPO/$CLI:master -h $ORIGIN_REPO/$CLI:$BRANCH
+
+  cd ..
+  echo ""
+  echo "PR opened against master to update version"
+  echo "MERGE THIS BEFORE CONTINUING"
+  echo ""
+}
+
+git_pull() {
+  cd $CLI
+  git pull
+  cd ..
+}
+
+sign() {
```

Comment:
I would skip the signing until there is a incubator wide process for determining who can sign code and what keys can be used.